tests/src/Functional/SettingsTest.php

```diff
@@ -0,0 +1,108 @@
+    $assert_session->fieldValueEquals('Disable TLS certificate check for COOL.', '');
+    $assert_session->fieldValueEquals('Allow COOL to use fullscreen mode.', '1');
+
+    // Check that fields values are saved.
+    $assert_session->fieldExists('Collabora Online server URL')
+      ->setValue('http://collaboraserver.com/');
+    $assert_session->fieldExists('WOPI host URL')
+      ->setValue('http://wopihost.com/');
+    $assert_session->fieldExists('JWT private key ID')
+      ->setValue('a_random_token');
+    $assert_session->fieldExists('Access Token Expiration (in seconds)')
+      ->setValue('3600');
+    $assert_session->fieldExists('Disable TLS certificate check for COOL.')
+      ->check();
+    // Since default is checked we disable the full screen option.
+    $assert_session->fieldExists('Allow COOL to use fullscreen mode.')
+      ->uncheck();
+    $assert_session->buttonExists('Save configuration')->press();
+    $assert_session->statusMessageContains('The configuration options have been saved.', 'status');
+
+    // Check the values in fields.
+    $this->drupalGet(Url::fromRoute('collabora-online.settings'));
+    $assert_session->fieldValueEquals('Collabora Online server URL', 'http://collaboraserver.com/');
+    // Slash is removed at the end of Wopi URL.
+    $assert_session->fieldValueEquals('WOPI host URL', 'http://wopihost.com');
+    $assert_session->fieldValueEquals('JWT private key ID', 'a_random_token');
+    $assert_session->fieldValueEquals('Access Token Expiration (in seconds)', '3600');
+    $assert_session->fieldValueEquals('Disable TLS certificate check for COOL.', '1');
+    $assert_session->fieldValueEquals('Allow COOL to use fullscreen mode.', '');
+
+    // Check required fields, set empty values and check errors displayed.
+    $this->drupalGet(Url::fromRoute('collabora-online.settings'));
+    $assert_session->fieldExists('Collabora Online server URL')->setValue('');
+    $assert_session->fieldExists('WOPI host URL')->setValue('');
+    $assert_session->fieldExists('JWT private key ID')->setValue('');
+    $assert_session->fieldExists('Access Token Expiration (in seconds)')->setValue('');
+    $assert_session->fieldExists('Disable TLS certificate check for COOL.')->uncheck();
+    $assert_session->fieldExists('Allow COOL to use fullscreen mode.')->uncheck();
+    $assert_session->buttonExists('Save configuration')->press();
+    $assert_session->statusMessageContains('Collabora Online server URL field is required.', 'error');
+    $assert_session->statusMessageContains('WOPI host URL field is required.', 'error');
+    $assert_session->statusMessageContains('JWT private key ID field is required.', 'error');
+    $assert_session->statusMessageContains('Access Token Expiration (in seconds) field is required.', 'error');
+    $assert_session->statusMessageNotContains('The configuration options have been saved.', 'status');
+
+    // Check validation for fields.
```

Comment:
For this I prefer 'check', matter of style, I prefer not to repeat/user the word 'test' in tests comments.

Reply:
Just to recap the conversation we had in chat:
"Check" is usually used to describe only the observation part. "Test" would describe a full circle of doing something and then observing the behavior.
E.g. "Check the brakes" would mean to have a look, but "Test the brakes" would mean to drive the car, hit the brake pedal, and then measure the performance.
So even if we are within the same test method, I think "test" is still the more suitable word if we describe a full sequence. Otherwise we can say "Do X" and then a separate comment "Observe Y" or just "Y happens".